eth_common/eth_types.nim

```diff
@@ -182,25 +185,26 @@ proc append*(rlpWriter: var RlpWriter, t: Time) {.inline.} =
 proc rlpHash*[T](v: T): Hash256 =
   keccak256.digest(rlp.encode(v).toOpenArray)
 
-template notImplemented =
+func blockHash*(h: BlockHeader): KeccakHash {.inline.} = rlpHash(h)
+
+proc notImplemented =
   assert false, "Method not impelemented"
 
 template deref*(r: BlockHeaderRef | BlockBodyRef): auto =
   r[]
 
-method genesisHash*(db: AbstractChainDb): KeccakHash =
-  notImplemented
+method genesisHash*(db: AbstractChainDb): KeccakHash {.base.} =
+  notImplemented()
```

Comment:
Will we need inheritance?

Reply:
Inheritance - dunno. The idea behind `AbstractChainDb` is the interface which is being used in `eth_p2p`, and implemented in nimbus.